UniWatch/Views/Lecture/Index.cshtml

```diff
@@ -7,13 +7,67 @@
 <h2>@Model.ClassName</h2>
 <hr />
 
-@{
+@if (Request.IsAuthenticated && User.IsInRole("Student"))
+{
```

Comment:
I largely agree with @the-pat's comments. You should not need to check for authentication or role in the veiw. A different route will be to have one index view file for teachers and another for students.